src/desktop.c

```diff
@@ -67,6 +67,18 @@ desktop_focus_view(struct view *view)
 	}
 
 	view_focus(view);
+
+	if (raise) {
+		view_move_to_front(view);
+	}
+
+	/*
+	 * Switch workspace if necessary to make the view visible
+	 * (unnecessary for "always on top" views).
+	 */
+	if (!view_is_always_on_top(view)) {
+		workspaces_switch_to(view->workspace, /*update_focus*/ false);
+	}
```

Comment:
Can we move this above the `view_focus()` line?
At least for me the mental model of
- switch to workspace where the view belongs to (e.g. "make it visible")
- focus the view
- maybe raise the view

is easier to follow than to focus / raise a potentially invisible view if its on a currently hidden workspace.

Reply:
Sure, fine by me